primitives/bridge/src/lib.rs

```diff
@@ -0,0 +1,385 @@
+};
+use bp_runtime::{Chain, ChainId, SourceAccount};
+use bridge_runtime_common::messages::{
+	source::{
+		FromThisChainMessagePayload, BAD_ORIGIN, OUTBOUND_LANE_DISABLED, TOO_LOW_FEE,
+		TOO_MANY_PENDING_MESSAGES,
+	},
+	AccountIdOf, BalanceOf, MessageBridge, ThisChain, ThisChainWithMessages,
+};
+use frame_support::{weights::Weight, Parameter};
+use sp_core::H256;
+use sp_runtime::{traits::Convert, RuntimeDebug};
+use sp_std::prelude::*;
+// --- darwinia-network ---
+use common_primitives::*;
+use darwinia_fee_market::RingBalance;
+
+/// Maximal size (in bytes) of encoded (using `Encode::encode()`) account id.
+pub const MAXIMAL_ENCODED_ACCOUNT_ID_SIZE: u32 = 32;
+
+/// Number of extra bytes (excluding size of storage value itself) of storage proof, built at
+/// Crab chain. This mostly depends on number of entries (and their density) in the storage trie.
+/// Some reserve is reserved to account future chain growth.
+pub const EXTRA_STORAGE_PROOF_SIZE: u32 = 1024;
+
+/// Number of bytes, included in the signed Crab transaction apart from the encoded call itself.
+///
+/// Can be computed by subtracting encoded call size from raw transaction size.
+pub const TX_EXTRA_BYTES: u32 = 103;
+/// Increase of delivery transaction weight on Crab chain with every additional message byte.
+///
+/// This value is a result of `pallet_bridge_messages::WeightInfoExt::storage_proof_size_overhead(1)` call. The
+/// result then must be rounded up to account possible future runtime upgrades.
+pub const ADDITIONAL_MESSAGE_BYTE_DELIVERY_WEIGHT: Weight = 25_000;
+/// Weight of single regular message delivery transaction on Crab chain.
+///
+/// This value is a result of `pallet_bridge_messages::Pallet::receive_messages_proof_weight()` call
+/// for the case when single message of `pallet_bridge_messages::EXPECTED_DEFAULT_MESSAGE_LENGTH` bytes is delivered.
+/// The message must have dispatch weight set to zero. The result then must be rounded up to account
+/// possible future runtime upgrades.
+pub const DEFAULT_MESSAGE_DELIVERY_TX_WEIGHT: Weight = 1_000_000_000;
+/// Maximal weight of single message delivery confirmation transaction on Crab chain.
+///
+/// This value is a result of `pallet_bridge_messages::Pallet::receive_messages_delivery_proof` weight formula computation
+/// for the case when single message is confirmed. The result then must be rounded up to account possible future
+/// runtime upgrades.
+pub const MAX_SINGLE_MESSAGE_DELIVERY_CONFIRMATION_TX_WEIGHT: Weight = 2_000_000_000;
+pub const PAY_INBOUND_DISPATCH_FEE_WEIGHT: Weight = 600_000_000;
+
+/// Maximal number of unrewarded relayer entries at inbound lane.
+pub const MAX_UNREWARDED_RELAYER_ENTRIES_AT_INBOUND_LANE: MessageNonce = 128;
+/// Maximal number of unconfirmed messages at inbound lane.
+pub const MAX_UNCONFIRMED_MESSAGES_AT_INBOUND_LANE: MessageNonce = 128;
+
+pub const DARWINIA_CRAB_LANE: [u8; 4] = *b"daca";
```

Comment:
@xiaoch05 @AsceticBear How about use  "00000000" as lane id for the first lane?

Reply:
I think it's ok. You mean 0x00000000 not "00000000"?

Reply:
> I think it's ok. You mean 0x00000000 not "00000000"?

Yes, hex format